test/distributed/shard_tensor/ops/test_knn.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import torch
+from torch.distributed.tensor.placement_types import Replicate, Shard
+
+from physicsnemo.distributed import DistributedManager, scatter_tensor
+from physicsnemo.utils.neighbors import knn
+
+from .utils import numerical_shard_tensor_check
+
+
+class kNNModule(torch.nn.Module):
+    def __init__(
+        self,
+        num_neighbors=4,
+    ):
+        super().__init__()
+
+        self.num_neighbors = num_neighbors
+
+    def forward(self, points, queries):
+        return knn(points, queries, self.num_neighbors)
+
+
+@pytest.mark.multigpu_static
+@pytest.mark.parametrize("scatter_points", [True, False])
+@pytest.mark.parametrize("scatter_queries", [True, False])
+def test_knn_1dmesh(
+    distributed_mesh,
+    scatter_points: bool,
+    scatter_queries: bool,
+):
```

Comment:
I'm not sure there is a use case for a 2D mesh of point clouds - since the points aren't sorted or organized, there is no logical way to do this.  I don't test 2D meshes, therefore.